cf-net/cf-net.c

```diff
@@ -0,0 +1,507 @@
+                Writer *w = FileWriter(stdout);
+                WriterWriteHelp(w, "cf-net", OPTIONS, HINTS, true);
+                FileWriterDetach(w);
+                exit(EXIT_SUCCESS);
+                break;
+            }
+            case 'd':
+            {
+                opts->debug = true;
+                break;
+            }
+            case 'v':
+            {
+                opts->verbose = true;
+                break;
+            }
+            case 'I':
+            {
+                opts->inform = true;
+                break;
+            }
+            case 'n':
+            {
+                opts->dry_run = true; // TODO: Not in use
+                break;
+            }
+            case 'V':
+            {
+                printf("cf-net version number "CF_NET_VERSION"\n");
+                exit(EXIT_SUCCESS);
+                break;
+            }
+            case 'i':
+            {
+                opts->interactive = true; // TODO: Not in use
+                break;
+            }
+            case 'l':
+            {
+                opts->timestamp = true; // TODO: Not in use
+                break;
+            }
+            case 'b':
+            {
+                opts->background = atoi(optarg); // TODO: Not in use
+                break;
+            }
+            case 't':
+            {
+                opts->timeout = atoi(optarg); // TODO: Not in use
+                break;
+            }
+            case 'M':
+            {
+                // TODO: How do we actually add a man page
+                Writer *out = FileWriter(stdout);
+                ManPageWrite(out, "cf-net", time(NULL),
+                             CF_NET_SHORT_DESCRIPTION,
+                             CF_NET_MANPAGE_LONG_DESCRIPTION,
+                             OPTIONS, HINTS,
+                             true);
+                FileWriterDetach(out);
+                exit(EXIT_SUCCESS);
+                break;
+            }
+            default:
+            {
+                break;
+            }
+        }
+    }
+    (*cmd) = argv[optind];
+    (*args) = &(argv[optind+1]);
+    return 0;
+}
+
+// Just for testing argument parsing
+int CFNetPrint(char **args)
+{
+    for (int i = 0; args[i] != NULL; ++i)
+    {
+        printf("%s ", args[i]);
+    }
+    printf("\n");
+    return 0;
+}
+
+void CFNetInit()
+{
+    CryptoInitialize();
+    LoadSecretKeys();
+    cfnet_init(NULL, NULL);
+}
+
+AgentConnection* CFNetOpenConnection(char *server)
+{
+
+    AgentConnection *conn = NULL;
+    {
+        ConnectionFlags connflags = {
+            .protocol_version = CF_PROTOCOL_LATEST,
+            .trust_server = true
+        };
+        int err = 0;
+        char *buf = xstrdup(server);
+        char *host, *port;
+        ParseHostPort(buf, &host, &port);
+        if (port == NULL)
+        {
+            port = CFENGINE_PORT_STR;
+        }
+        conn = ServerConnection(host, port, 30, connflags, &err);
```

Comment:
I believe this call would modify the trust database (i.e. `ppkeys` directory) and lastseen entries of the whole CFEngine installation, which is bad for a test tool. I'm not if this is hard to fix, if it is add a TODO note because it's important.